### PR TITLE
build: update to version of neutrino w/ BIP 155 support

### LINKED ISSUE #4973: Enable Tor v3 address support for Neutrino client mode
### Background

Tor v3 address are not supported in Neutrino client mode.  This is discussed in the first comment here:

https://github.com/lightningnetwork/lnd/pull/4806

This is more relevant as bitcoind recently released version 0.21.0 which adds support for serving compact filters to peers.  Bitcoind also switched to Tor v3 addresses as a default in version 0.21.0 (older versions only supported v2).  Enabling Tor v3 address support could increase the number of Neutrino mode LND nodes deployed as there would be many more backends available.

### PR DESCRIPTION
**Pull request by Roasbeef**
Fixes #4973

### CODE REVIEW

**Comment by Roasbeef** [comment]
Weird that the go.sum didn't change....

**Comment by Crypt-iQ** [comment]
so there's a bit more that needs to be done: the dialer and nameresolver for neutrino have to change

**Comment by guggero** [comment]
It looks like the tag `v0.14.0` of neutrino isn't for a commit on the master branch...

**Comment by Roasbeef** [comment]
/facepalm 

correcting...

**Comment by Crypt-iQ** [comment]
Here's the patch I used to sync neutrino on mainnet w/ bip155 & lnd - everything but the modules changes are relevant: https://github.com/Crypt-iQ/lnd/tree/neutrino_config

**Comment by Roasbeef** [comment]
Updated: new tag, release notes, the diff above applied.

File: docs/release-notes/release-notes-0.15.0.md [hunk 1]
```diff
@@ -44,6 +44,12 @@ then watch it on chain. Taproot script spends are also supported through the
 * Add [--socksproxy](https://github.com/lightningnetwork/lnd/pull/6422)
   to allow for RPC calls via Tor.
 
+## Neutrino
+
+[Neutrino now suports BIP
+155](https://github.com/lightningnetwork/lnd/pull/6468), allowing it to connect
+to Bitcoin nodes that advertise a Tor v3 onion service address.
```
Comment (Crypt-iQ) [inline docs/release-notes/release-notes-0.15.0.md:51]: It could previously connect, just couldn't gossip

**Review by Crypt-iQ — APPROVED** [review]
LGTM

File: docs/release-notes/release-notes-0.15.0.md (hunk 1 above)
Comment (hsjoberg) [reply docs/release-notes/release-notes-0.15.0.md:51]: @Crypt-iQ No, it's currently not possible for Neutrino to connect to a v3 service.

The error message is
```
[WRN] BTCN: unable to lookup IP for 0230a5bca558e6741460c13dd34e636da28e52afd91cf93db87ed1b0392a7466eb@nwozjjgxilimzjd3d23vfgkmdsr6j2wekia4d77f673cefrfot6wtqqd.onion:9735: tor general error
```

**Pull request merged**